src/runtime/task/mod.rs

```diff
@@ -89,7 +90,16 @@ pub(crate) struct Task {
     name: Option<String>,
 
     local_storage: StorageMap,
-    pub(super) span: tracing::Span,
+
+    // The `Span` containing the `Task`s `id` and the current step count (if step count recording is enabled)
+    pub(super) step_span: Span,
+    // The current `Span` "stack" of the `Task`. There are two things to note:
+    // 1: We have to own the `Span`s (versus storing `Id`s) for the `Span` to not get dropped while the task is switched out.
+    // 2: We have to store the stack of `Span`s in order to return to the correct `Span` once the `Entered<'_>` from an
+    //    `instrument`ed future is dropped.
+    // `span_stack` is (as the name suggests) a stack. We `pop` it empty when resuming a `Task`, and `push` + `exit`
```

Comment:
Also add a comment here stating the order in which spans are stored in the `span_stack` (the outermost span is at the top of the stack)